sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/implementation/util/ModelHelper.java

```diff
@@ -57,6 +58,8 @@ public static ParallelTransferOptions populateAndApplyDefaults(ParallelTransferO
                 : other.getNumBuffers(),
             other.getProgressReceiver(),
             other.getMaxSingleUploadSize() == null ? Integer.valueOf(MAX_APPEND_FILE_BYTES)
-                : other.getMaxSingleUploadSize());
+                : other.getMaxSingleUploadSize(),
+            // Queues.SMALL_BUFFER_SIZE is the default used by reactor
+            other.getMaxConcurrency() == null ? Integer.valueOf(Queues.SMALL_BUFFER_SIZE) : other.getMaxConcurrency());
```

Comment:
Would we want to use a parallelization that is more idiomatic to the Storage service usage? I know `Queues.SMALL_BUFFER_SIZE` aligns more with Reactor's usage but could the higher level of parallelization get use into issues.

Reply:
Fair point. It is true that accepting this default is what hurt one the customer who filed the issue and why we're making this change. But that also seems like a fairly rare case so far? And I'm not sure what a more idiomatic value for Storage would be. It feels like anything we choose would just be a guess.

Reply:
https://github.com/reactor/reactor-core/blob/6058a391f614de6213fb85970272fc5b342bd181/reactor-core/src/main/java/reactor/util/concurrent/Queues.java#L88

Looks like that'd be 256 by default. Pretty high.

We limit number of buffers. Shouldn't max concurrency == numBuffers ? Or maxConcurrency be <= numbBuffers with numBuffers being default ?